core/scripts/parsers/board_parser.py

```diff
@@ -0,0 +1,41 @@
+"""@package parsers
+Parses the boards.txt file and gathers information about the current board
+"""
+
+from core.scripts.others import helper
+
+
+def get_board_properties(board, board_file_path):
+    """parses the board file returns the properties of the board specified"""
+
+    board_file = open(helper.linux_path(board_file_path))
+    board_str = board_file.readlines()
+
+    board_id_str = mcu_str = f_cpu_str = board_name_str = ""
+
+    for line in board_str:
+        if board in line:
+            if "board=" in line:
+                board_id_str = line[line.find('='):].strip("=").strip("\n").strip(" ")
+            elif "mcu=" in line:
+                mcu_str = line[line.find('='):].strip("=").strip("\n").strip(" ")
+            elif "f_cpu=" in line:
```

Comment:
I think it might be better to parse the whole `boards.txt` as a tree, i.e. `boards[boardName]['build']['f_cpu']` and cache it.